### PR TITLE
deid export downloads from daily saved

### PR DESCRIPTION
**Pull request by benrudolph**
@czue @millerdev holy cow this ended up being way more complicated than i originally thought. there's a lot of indirection going on, but this should fix. more like a bandaid; this code will be gone soon. pretty difficult to write tests for because this is pretty intense view logic, but open to push back. was going to take me a while to do, so thought it'd be good to at least get the fix in

http://manage.dimagi.com/default.asp?225223

### CODE REVIEW

File: corehq/apps/reports/urls.py [hunk 1]
```diff
@@ -97,6 +97,7 @@
     url(r"^export/bulk/download/$", "export_default_or_custom_data", name="export_bulk_download", kwargs=dict(bulk_export=True)),
     # saved
     url(r"^export/saved/download/(?P<export_id>[\w\-]+)/$", "hq_download_saved_export", name="hq_download_saved_export"),
+    url(r"^export/saved/download/deid/(?P<export_id>[\w\-]+)/$", "hq_deid_download_saved_export", name="hq_deid_download_saved_export"),
```
Comment (dimagimon) [inline corehq/apps/reports/urls.py]: E501 line too long (136 > 115 characters)

File: corehq/apps/reports/views.py [hunk 2]
```diff
@@ -520,6 +522,21 @@ def _export_default_or_custom_data(request, domain, export_id=None, bulk_export=
...
+@require_form_deid_export_permission
+@require_GET
+def hq_deid_download_saved_export(req, domain, export_id):
+    export = SavedBasicExport.get(export_id)
+    if not export.is_safe:
+        raise Http404()
+    return _download_saved_export(req, domain, export)
+
+
+def _download_saved_export(req, domain, export):
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: F811 redefinition of unused 'export' from line 144

File: corehq/apps/reports/views.py [hunk 3]
```diff
@@ -520,6 +522,21 @@ def _export_default_or_custom_data(request, domain, export_id=None, bulk_export=
...
     export = SavedBasicExport.get(export_id)
+    return _download_saved_export(req, domain, export)
+
+
+@csrf_exempt
+@login_or_digest_or_basic_or_apikey(default='digest')
+@require_form_deid_export_permission
+@require_GET
+def hq_deid_download_saved_export(req, domain, export_id):
+    export = SavedBasicExport.get(export_id)
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: F811 redefinition of unused 'export' from line 144

File: corehq/apps/reports/views.py [hunk 4]
```diff
@@ -67,7 +67,7 @@
     SchemaMismatchException
 )
 from couchexport.export import Format, export_from_tables
-from couchexport.models import DefaultExportSchema, SavedBasicExport
+from couchexport.models import DefaultExportSchema, SavedBasicExport, SavedExportSchema
```
Comment (dimagimon) [inline corehq/apps/reports/views.py]: F401 'SavedExportSchema' imported but unused

File: corehq/apps/reports/views.py (hunk 3 above)
Comment (benrudolph) [inline corehq/apps/reports/views.py]: haha there's an import named `export` as well...

**Comment by millerdev** [comment]
The fogbugz case URL looks incorrect

**Comment by millerdev** [comment]
It's not clear to me what this "new feature?" is supposed to do. After review none of the code looks obviously wrong, but it seems to require quite some context to understand what it affects.

**Comment by benrudolph** [comment]
oops forgot to put a summary. this basically makes it so deid users are able to download daily saved deid exports. proper ticket here: http://manage.dimagi.com/default.asp?225223#1134280

**Pull request merged**